test/memory.test.js

```diff
@@ -502,6 +503,14 @@ describe('Memory connector', function() {
       });
     });
 
+    it('should support nested date property in query', function(done) {
+      User.find({where: {'address.occupiedSince': new Date('2016-01-01')}},
+        function(err, users) {
+          should(users.length).be.equal(1);
```

Comment:
When an assert like this fails, it's hard to tell what went wrong, because the error message provides very little details.

A better way is to assert on some property of the returned users, e.g. name.

```js
var actualNames = users.map(function(u) { return u.name });
should(actualNames).be.eql(['John Lennon']);
```